content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+     
+        int m_unboundedArrayOfIntegers[]; // ERROR. Unbounded arrays are not supported for fundamental(packables, in general) data types.
+     
+        Texture2D<float4> m_textures[MyArraySize]; // GOOD. Contant size array of Resource View types is ok.
+         
+        ConstantBuffer<SomeStruct> m_unboundedArrayOfStructs[]; // GOOD. Unbounded array of Resource View type is ok. BUT, there are some restrictions.
+    }
+  
+To learn more about what are the limitations when declaring unbounded arrays, go here: [Binding Rules For Unbounded Arrays.](unbounded-arrays)  
+  
+### <a name="partial"></a> Partial ShaderResourceGroup Definitions.
+Before talking about `partial ShaderResourceGroup`, it's important to talk about [ShaderResourceGroupSemantic](srg-semantics), which is always required when declaring a `ShaderResourceGroup`.  
+Go to this section to read about [ShaderResourceGroupSemantic](srg-semantics).  
+  
+The `partial` keyword is useful to define parts of an SRG in different blocks. For example, across different files. This is a similar concept as partial classes in C#.  
+There are no limits into how many `partial` blocks can be used to define an SRG.  
+There are only three requirements:  
+1. When using `partial` to define an SRG block, all other SRG definition blocks, for the same SRG, must start with the `partial` keyword.
+2. For a given SRG, at least one `partial` SRG block must define the Shader Resource Group Semantic.
+3. For a given SRG, if the Shader Resource Group Semantic is defined more than once across different `partial` blocks, they must refer to the same `ShaderResourceGroupSemantic`.  
+  
+**Example: Using the partial keyword**  
+  
+    ShaderResourceGroupSemantic PerPass
+    {
+        FrequencyId = 1;
+    }
+     
+    partial ShaderResourceGroup PassSrg
+    {
+        Texture2D m_texture; // A regular SRV resource,
+    }
+     
+    partial ShaderResourceGroup PassSrg : PerPass // At least one partial block must define the semantic
+    {
+        Sampler m_sampler; // A Dynamic sampler.
+    }
+     
+    partial ShaderResourceGroup PassSrg
+    {
+        float2 m_resolution;
+        float4 ReadColor(float2 uv)
+        {
+            return m_texture.Sample(m_sampler, uv);
+        }
+    }
+     
+    PSOutput MainPS(VSOutput IN)
+    {
+        PSOutput OUT;
+        float2 uv = IN.m_position.xy / PassSrg::m_resolution;
+        OUT.m_color = PassSrg::ReadColor(uv);
+        return OUT;
+    }
+  
+An important remark about partial SRG blocks is that when the compiler, **AZSLc**, finds the first `partial` block, it will use it as the unified point of emission for all the data of a given SRG across all `partial` blocks. The main recommendation that comes out of this rule, is that any globally defined `struct`, or `class` that may be referenced by a `partial` SRG block must be defined before the first partial block for the SRG in question.  
```

Comment:
See suggestion above. Move this before the first example, so it doesn't get lost amidst the series of examples.
```suggestion
```

Reply:
Fixed.